lib/controllers/v1/computervision_controller.js

```diff
@@ -61,48 +62,64 @@ var ComputervisionController = class ComputervisionController {
       if ( !response || _.isEmpty( response.results ) ) {
         return callback( { message: "Unknown observation" } );
       }
-      const obs = response.results[0];
+      const observation = response.results[0];
       let photoURL;
-      _.each( obs.photos, p => {
+      _.each( observation.photos, p => {
         if ( photoURL ) { return; }
         if ( p.url && p.url.match( /static\.inaturalist.*\/square\.jpe?g[\?$]/i ) ) {
           photoURL = p.url.replace( "/square.", "/medium." );
         }
       });
       if ( !photoURL ) { return callback( { message: "Observation has no scorable photos" } ); }
-      // download the JPG
-      request.head( photoURL, ( err, res ) => {
-        if ( err ) { return callback( err ); }
-        if ( !res.headers[ "content-type" ].match( /image\/jpe?g/ ) ) {
-          return callback( { message: "There was a problem scoring this photo" } );
-        }
-        const tmpFilename = `observation-${obsID}.jpg`;
-        const tmpPath = path.resolve( global.config.imageProcesing.uploadsDir, tmpFilename );
-        request( photoURL ).pipe( fs.createWriteStream( tmpPath ) ).on( "close", ( ) => {
-          let scoreImageReq = Object.assign( req, {
-            file: {
-              filename: tmpFilename,
-              mimetype: "image/jpeg"
-            }
-          });
-          if ( !scoreImageReq.body ) { scoreImageReq.body = { }; }
-          if ( !scoreImageReq.body.lat && obs.location ) {
-            const latLng = obs.location.split( "," );
-            scoreImageReq.body.lat = latLng[0];
-            scoreImageReq.body.lng = latLng[1];
-          }
-          if ( !scoreImageReq.body.observed_on && obs.observed_on ) {
-            scoreImageReq.body.observed_on = obs.observed_on;
+      req.query.image_url = photoURL;
+      ComputervisionController.scoreImageURL( req, { observation }, callback );
+    });
+  }
+
+  static scoreImageURL( req, options, callback ) {
+    options = options || {};
+    if( !req.userSession && !req.applicationSession ) {
+      return callback({ error: "Unauthorized", status: 401 });
+    }
+    const photoURL = req.query.image_url;
+    if ( !photoURL ) {
+      return callback( { message: "No scorable photo", status: 422 } );
+    }
+    // download the JPG
+    request.head( photoURL, ( err, res ) => {
+      if ( err ) { return callback( err ); }
+      if ( !res.headers[ "content-type" ].match( /image\/jpe?g/ ) ) {
```

Comment:
We can remove the JPEG check now, and just make sure it's some kind of `image/*`. Not necessary to to in this commit, we can do it later, but just FYI the python app will be attempting conversion now